dev/src/MockGRPCManager.ts

```diff
@@ -106,8 +116,27 @@ export class MockGrpcManager {
     public createCourse(course: Course): Promise<IGrpcResponse<Course>> {
         let data: Course | null = null
         const found = this.courses.getCoursesList().find(c => c.getId() === course.getId())
-        if (!found || !this.currentUser.getIsadmin()) {
+        const isAdmin = this.currentUser?.getIsadmin()
+        const user = this.currentUser
+        if (!found && user && isAdmin) {
+            // TODO: This is not a guaranteed unique ID
```

Comment:
What would be the solution to producing a unique ID here and other places?

I'm not sure I quite understand why the IDs must/should be unique, but could a solution be to use a map from some string key (e.g., `CourseID`) that points to another map of already used IDs that a `generateID("CourseID")` function should check before returning a new ID? (Or something along those lines???)

Reply:
The current solution using `<Array>.length + 1` causes issues if an element in the array is deleted, and a new element is inserted, example:

`Courses -> [{courseID: 1}, {courseID: 2}, {courseID: 3}]`.

Delete course with ID == 1 -> `[{courseID: 2}, {courseID: 3}]`
Insert new course (ID is set to `courseList.length + 1`) -> `[{courseID: 2}, {courseID: 3}, {courseID: 3]`

Which will cause issues when mapping objects such as assignments to their related course by their course ID.

Your suggested solution sounds good.

Reply:
Understand. Suggest to implement my solution to avoid problems down the line. I think it will be possible to do it with a separate function that works across different IDs.

Reply:
It might be a good thing to add a unit test for such a `generateID()` function.